src/query_executor.c

```diff
@@ -169,10 +169,12 @@ void MGraph_AcquireReadLock() {
 
 void MGraph_AcquireWriteLock(RedisModuleCtx *ctx) {
     pthread_rwlock_wrlock(&_rwlock);
+    _writelocked = true;
```

Comment:
@swilly22 I admit that this does feel a little dodgy to me, but I could not find any POSIX function that would inform a thread whether it was holding a read or write lock. This variable is accessed on module load, and subsequently only by a writer while it is holding a lock.

Reply:
I've tried to perform a similar search, but came up with nothing...